platform/linux-generic/include/odp_ring_mpmc_internal.h

```diff
@@ -154,8 +154,8 @@ static inline uint32_t _RING_MPMC_DEQ_MULTI(_ring_mpmc_gen_t *ring,
 	 * and thus r_head value is always behind or equal to w_tail value.
 	 * When CAS operation succeeds, this thread owns data between old
 	 * and new r_head. */
+	old_head = odp_atomic_load_acq_u32(&ring->r.r_head);
 	do {
-		old_head = odp_atomic_load_acq_u32(&ring->r.r_head);
```

Comment:
The CAS operation loads r_head using relaxed memory ordering while this atomic load used acquire memory ordering. So moving this load out of the loop changes memory ordering. Without thinking the code through but by just reading the comment above ("Load acquires ensure...") it appears that the change in memory ordering could break things. Even if the change was correct, the commit message should comment the change in memory ordering.

Reply:
Yes. If this change is done, then the CAS operation should be changed to "load acquire on failure" as well.

Reply:
Sorry, it was my mistake not noticing the memory order of CAS operation. Since the modification doesn't improve performance, I guess it's better to leave it as is. 
How can I close/abandon this PR?

Reply:
You can close a pull request by clicking the "Close pull request" button below.